### PR TITLE
Pass MF2WG string test #02

### PR DESCRIPTION
**Pull request by mvilks**
Collecting non-critical errors, e.g., when input variable not defined, it generates an error `unresolved-variable` but also executes the template using some fallback value (nil).

### CODE REVIEW

File: template/template.go [hunk 1]
```diff
@@ -112,6 +112,11 @@ type executer struct {
 	template  *Template
 	w         io.Writer
 	variables map[string]any
+	errs      []error
```
Comment (janishorsts) [inline template/template.go:115]: not needed?

File: .message-format-wg
Comment (janishorsts) [inline .message-format-wg:1]: changing MF2 submodule commit is scope creep

File: template/template.go [hunk 2]
```diff
@@ -424,7 +431,11 @@ func (e *executer) resolveSelector(matcher ast.Matcher) ([]any, error) {
 		}
 
 		input, err := e.resolveValue(selectExpr.Operand)
-		if err != nil {
+		if errors.Is(err, mf2.ErrUnresolvedVariable) {
```
Comment (janishorsts) [inline template/template.go:434]: This does not align with my comment on the ticket - how to handle errors in selectors from the MF2 spec.

Comment (janishorsts) [reply template/template.go:434]: Should `resolveValue()` return `ErrUnresolvedVariable` instead? Resolving a variable can occur in different places in the MF2. I understand `ErrUnresolvedVariable` should be returned everywhere we try to resolve a variable.

File: template/template.go (hunk 1 above)
Comment (mvilks) [reply template/template.go:115]: It is needed to collect possible errors from several selectors

File: .message-format-wg
Comment (mvilks) [reply .message-format-wg:1]: reverted

File: mf2wg_test.go [hunk 3]
```diff
@@ -260,7 +258,7 @@ func assertErr(t *testing.T, want Errors, err error) {
 
 	wantErr := func(want error) {
 		if !errors.Is(err, want) {
-			t.Errorf("want '%s', got '%s'", want, err)
+			t.Errorf("want error %s, got %s", want, err)
```
Comment (janishorsts) [inline mf2wg_test.go:261]: where did single quotes go? :)

**Review by janishorsts — APPROVED** [review]
